pkg/config/options.go

```diff
@@ -77,3 +130,17 @@ func (o *ConfirmableOptions) Bind(fs *flag.FlagSet) {
 	o.Options.Bind(fs)
 	fs.BoolVar(&o.Confirm, "confirm", false, "Create the branched configuration files.")
 }
+
+func getUpstreamBranch() string {
+	cmd := exec.Command("git", "rev-parse", "--abbrev-ref", "--symbolic-full-name", "@{upstream}")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
```

Comment:
Probably better to check for an exact error - what if this fails because there's no `git` in the system?

Reply:
yea if it fails we use `master` as a value. Also if there is no `git` in the system, we will fail later on either way. But I don't think that the actual error is relevant here. We try to get the upstream branch, if not, use `master`.

Reply:
I added a check to look for git command in the system [here](https://github.com/openshift/ci-tools/pull/2137/commits/e7454527ca93589780fa84854d686a705d705fed#diff-6731fb6f709392ce3e37d3b0c42074cddbce566dad2bab86af24ba7585eeb57cR56-R60)

Reply:
I am making a comment about this error check - you want to fall back to `master` in the very specific case that there is no upstream tracking branch configured. Check for that error instead of swallowing everything.

Reply:
addressed